### ISSUE TITLE
Tunnelled/proxied SSH connections

### ISSUE THREAD

**Issue by DeeHants** [SchoofsKelvin/vscode-sshfs#4]
This plugin looks great, however our organisation's setup requires that all our SSH connections go via a jump host.
This is setup with a PuTTY connection to the jump host that has a "dynamic" (SOCKS) tunnel configured.
Each PuTTY connection to the actual servers is then set to use a SOCKS proxy on localhost.

It looks like this plugin doesn't yet honour the proxy setting as it tries to connect directly to the specified hostname.

Supporting this would make things massively easier for my workflow.

**Comment by DeeHants** [comment]
I've also raised this on the ssh2 project:
https://github.com/mscdex/ssh2/issues/694

**Comment by SchoofsKelvin** [comment]
I've seen that ssh2 does allow "taking over" already established connections instead of creating them themselves. I'll be looking into this, as it seems like something that I could implement without too much trouble.

**Comment by DeeHants** [comment]
Cool, I saw it could use an existing sock so I was investigating an intermediary socks module too.

**Comment by SchoofsKelvin** [comment]
I've added a new, although experimental feature to use SOCKS proxies in version 1.5.0 which is also published to the marketplace.

For testing it, I had this setup:
* One PuTTY session "test" to test-server with a dynamic tunnel (D9159)
* Second PuTTY session "test-proxy" to localhost:22 using localhost:9159 as proxy
* Using the following configuration:
```json
{
  "putty": "test-proxy",
  "label": "Test - Proxy",
  "name": "test-proxy"
}
```
* Start the first session from PuTTY to establish the dynamic tunnel
* Connect to "test-proxy" in VSCode, which'll read PuTTY and fill in the full configuration:
```jsonc
{
    "putty": "test-proxy",
    "label": "Test - Proxy",
    "name": "test-proxy",
    // host, port, keyfile, agent, ...
    // Settings below are read/interfered from the PuTTY session
    // If you don't have a PuTTY session to read these from, you can add them manually
    // (but the JSON schema doesn't have these new fields yet, so won't suggest them)
    "proxy": {
        "type": "socks4", // or "socks5"
        "host": "localhost",
        "port": 9159,
    }
}
```
* The `socks` module requires an IP for the proxy, so "localhost" gets resolved to `127.0.0.1`
* The `socks` module creates a connection to (the remote) localhost:22 using the SOCKS tunnel
* Do a quick test to see if there any problems with reading/writing directies/files using the proxy

I've ran a second test using another SSH server as proxy, and connecting through that server, using `IP:22` instead of `localhost:22`, since the proxy server isn't the targeted server.

If you find any bugs, feel free to post them here, or make a new issue. I'll leave this one open for a bit.
